File: Chapter05/Java/StorageExample1/app/src/main/java/net/zenconsult/mobile/StoargeExample1Activity.java
package net.zenconsult.mobile;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.widget.EditText;

import java.util.Hashtable;

public class StoargeExample1Activity extends AppCompatActivity {

    private static final String TAG = "StoargeExample1Activity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @Override
    protected void onResume() {
        super.onResume();

        Hashtable<String, Object> data = new Hashtable<>();
        data.put("hostname", "smtp.gmail.com");
        data.put("port", 587);
        data.put("ssl", true);

        if (StoreData.storeData(data, this)) {
            Log.i(TAG, "onResume=>Successfully wrote data");
        } else {
            Log.e(TAG, "onResume=>Failed to write data to Shared Prefs");
        }

        EditText et = findViewById(R.id.editText1);
        et.setText(RetrieveData.get(this).toString());
    }
}

File: Chapter05/Kotlin/StorageExample1/app/src/main/java/net/zenconsult/mobile/StoargeExample1Activity.kt
package net.zenconsult.mobile

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.EditText
import java.util.*

class StoargeExample1Activity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    override fun onResume() {
        super.onResume()
        val data = Hashtable<String, Any?>()
        data["hostname"] = "smtp.gmail.com"
        data["port"] = 587
        data["ssl"] = true

        if (StoreData.storeData(data, this)) {
            Log.i(TAG, "onResume=>Successfully wrote data")
        } else {
            Log.e(TAG, "onResume=>Failed to write data to Shared Prefs")
        }

        val ed = findViewById<EditText>(R.id.editText1)
        ed.setText(RetrieveData.get(this).toString())
    }

    companion object {
        const val TAG = "StoargeExample1Activity"
    }
}